oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -151,6 +158,49 @@ public AccessToken refreshAccessToken() throws IOException {
     return new AccessToken(accessToken, new Date(expiresAtMilliseconds));
   }
 
+  /**
+   * Returns an Google Id Token from the metadata server on ComputeEngine.
+   * 
+   * @param targetAudience The aud: field the IdToken should include.
+   * @param options        List of Credential specific options for for the
+   *                       token. For example, an IDToken for a
+   *                       ComputeEngineCredential could have the full formated
+   *                       claims returned if
+   *                       ComputeCredentials.ID_TOKEN_FORMAT_FULL) is provided as
+   *                       a list option.  Valid option values are:
+   *                       * ComputeCredentials.ID_TOKEN_FORMAT_FULL<br>
+   *                       * ComputeCredentials.ID_TOKEN_LICENSES_TRUE<br>
+   *                       If no options are set, the default
+   *                       are "&amp;format=standard&amp;licenses=false"
+   * @throws IdTokenProvider.IdTokenProviderException if the attempt to get an
+   *                                                  IdToken failed
+   * @return IdToken object which includes the raw id_token, JsonWebSignature.
+   */
+  @Override
+  public IdToken idTokenWithAudience(String targetAudience, List<String> options) {
+    String optionalParams = "";
+    if (options != null) {
+      if (options.contains(ID_TOKEN_FORMAT_FULL))
+        optionalParams = "&format=full";
+      if (options.contains(ID_TOKEN_LICENSES_TRUE))
+        optionalParams = optionalParams + "&licenses=TRUE";
+    }
+    String documentURL = getIdentityDocumentUrl() + "?audience=" + targetAudience + optionalParams;
```

Comment:
`GenericUrl` should be used to build up this url instead of concatenating strings.

```java
GenericUrl documentUrl = new GenericUrl(getIdentityDocumentUrl());
documentUrl.set("audience", targetAudience);
// set other optional params
```

Reply:
added in as url